op-node/rollup/driver/state.go

```diff
@@ -223,7 +234,12 @@ func (s *Driver) eventLoop() {
 				}
 			}
 			planSequencerAction() // schedule the next sequencer action to keep the sequencing looping
-		case payload := <-s.unsafeL2Payloads:
+		// TODO: Should this be lower-priority in the switch case?
+		case <-altSyncTicker.C:
+			// Check if there is a gap in the current unsafe payload queue. If there is, attempt to fetch
+			// missing payloads from the backup RPC (if it is configured).
+			s.checkForGapInUnsafeQueue(ctx)
+		case payload := <-s.UnsafeL2Payloads:
```

Comment:
This is a great question- I don't believe so. There's also the possibility that it takes longer than 15 seconds to fill the gap, meaning that this method could also add duplicate payloads to the `UnsafeL2Payloads` channel (alongside P2P). @protolambda @trianglesphere, what do you think? Mark suggested turning off P2P while backup syncing is active, but that still leaves the possibility that backup sync takes longer than expected & adds duplicate blocks to the `PayloadsQueue`.

Reply:
Do we have deduplication somewhere in the existing P2P stack? I guess it might be at the p2p edge. We should make sure that the unsafe payload queue can handle multiple / late unsafe blocks

Reply:
Switching off p2p sync while we are doing RPC sync seems really tricky & error prone

Reply:
> Do we have deduplication somewhere in the existing P2P stack? I guess it might be at the p2p edge. We should make sure that the unsafe payload queue can handle multiple / late unsafe blocks

Not sure, will look into this today!

Reply:
libp2p deduplicates messages and has been configured with a seenMessageTTL that's long enough for it to remember a gossiped block for the entire period that it's considered valid gossip. However, that mechanism wouldn't (currently) know about any blocks you'll pulled via RPC sync you may get those duplicated.